src/core/dbus-execute.c

```diff
@@ -3140,6 +3141,36 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "ExecSearchPath")) {
+                _cleanup_strv_free_ char **l = NULL;
+                char **p;
+
+                r = sd_bus_message_read_strv(message, &l);
+                if (r < 0)
+                        return r;
+
+                STRV_FOREACH(p, l) {
+                        if (!path_is_absolute(*p) || strchr(*p, ':'))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid %s", name);
+                }
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        if (strv_isempty(l)) {
+                                strv_free(c->exec_search_path);
```

Comment:
you free the field here, but leave it set. you need to unset it too. To make this easy strv_free() (like most our destructor functions) returns NULL, properly cast. i.e write this:

```
c->exec_search_path = strv_free(c->exec_search_path);
```